test/test-helpers.h

```diff
@@ -92,9 +72,30 @@ class TempFile {
 				throw tempfileexception(strerror(saved_errno));
 			}
 		}
-	};
+	}
 
-	void init_file()
+	~MainTempDir()
+	{
+		::rmdir(tempdir.c_str());
```

Comment:
It could fail, but we can't know if that's normal or not. Consider code like this:

```c++
TempFile first;
{
    TempFile second;
}
std::ofstrem out(first.getPath());
```

`TempFile` incorporates a `MainTempDir` object. When `second` goes out of scope and is destroyed, we actually *don't* want it to remove the whole tempdir, otherwise `out` will try to create a file in a non-existent directory.

Perhaps I should add a comment saying: "Try to remove the tempdir, but don't try *too* hard: there might be other objects still using it. The last one will hopefully delete the tempdir". What do you think?

Reply:
I've so far never touched newsboats test infrastructure, and the comment would definitly have helped me there, and the reasoning seems fine to me.